src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java

```diff
@@ -70,6 +84,76 @@ public static void initIndexes() throws Exception {
         Collections.sort(indexes);
     }
     
+    @AfterClass
+    public static void tearDownStatics() {
+        indexes = null;
+        indicesDir = null;
+    }
+    
+    @Override
+    public Settings nodeSettings(int ord) {
+        return ImmutableSettings.builder()
+            .put(Node.HTTP_ENABLED, true) // for _upgrade
+            .put(MergePolicyModule.MERGE_POLICY_TYPE_KEY, NoMergePolicyProvider.class) // disable merging so no segments will be upgraded
+            .build();
+    }
+    
+    void setupDataLoadingNodes() throws Exception {
+        Path dataDir = newTempDirPath(LifecycleScope.SUITE);
+        indicesDir = dataDir.resolve(internalCluster().getClusterName()).resolve("nodes/0/indices/");
+        assertFalse(Files.exists(indicesDir));
+        Files.createDirectories(indicesDir);
+        ImmutableSettings.Builder nodeSettings = ImmutableSettings.builder()
+            .put("path.data", dataDir.toAbsolutePath());
+        internalCluster().startNode(nodeSettings);
+    }
+
+    String loadIndex(String indexFile) throws Exception {
+        Path unzipDir = newTempDirPath();
+        Path unzipDataDir = unzipDir.resolve("data");
+        String indexDir = indexFile.replace(".zip", "");
+        String indexName = indexDir.toLowerCase(Locale.ROOT);
+        
+        // decompress the index
+        Path backwardsIndex = Paths.get(getClass().getResource(indexFile).toURI());
+        try (InputStream stream = Files.newInputStream(backwardsIndex)) {
+            TestUtil.unzip(stream, unzipDir);
+        }
+        
+        // check it is unique
+        assertTrue(Files.exists(unzipDataDir));
+        Path[] list = FileSystemUtils.files(unzipDataDir);
+        if (list.length != 1) {
+            throw new IllegalStateException("Backwards index must contain exactly one cluster");
+        }
+        
+        // move to the real data dir
+        Path src = list[0].resolve("nodes/0/indices/" + indexName);
+        Path dest = indicesDir.resolve(indexName);
+        /*StringBuffer msg = new StringBuffer();
+        msg.append("[" + indexFile + "] missing index dir: " + src.toString() + "\nContents:\n");
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(src.getParent())) {
+            for (Path entry: stream) {
+                msg.append(entry.toString() + "\n");
+            }
+        }
+        assertTrue(msg.toString(), Files.exists(src));
+        */assertTrue("[" + indexFile + "] missing index dir: " + src.toString(), Files.exists(src));
+        Files.move(src, dest);
+        assertFalse(Files.exists(src));
+        assertTrue(Files.exists(dest));
+        
+        // force reloading dangling indices with a cluster state republish
+        client().admin().cluster().prepareReroute().get();
+        ensureGreen(indexName);
```

Comment:
I think there is a race condition here - it may take some time for the dangling logic to kick in (it's async via network calls). I would assertBusy until the index is visible in the cluster state, then go into ensure green.